tools/pipelines/templates/build-npm-package.yml

```
@@ -407,6 +412,41 @@ extends:
                         inputs:
                           sourceFolder: '${{ parameters.buildDirectory }}/nyc/report'
                           targetFolder: $(Build.ArtifactStagingDirectory)/codeCoverageAnalysis
+                      - task: Bash@3
+                        displayName: Report Code Coverage Comparison
+                        condition: and(succeededOrFailed(), eq('${{ parameters.reportCodeCoverageComparison }}', true), eq(variables['ReportDirExists'], 'true'), eq(variables['Build.Reason'], 'PullRequest'), eq(variables['System.PullRequest.TargetBranch'], 'main'))
+                        continueOnError: false
+                        env:
+                          ADO_API_TOKEN: $(System.AccessToken)
+                          GITHUB_API_TOKEN: $(githubPublicRepoSecret)
+                          TARGET_BRANCH_NAME: '$(targetBranchName)'
+                          ADO_BUILD_ID: $(Build.BuildId)
+                        inputs:
+                          targetType: inline
+                          workingDirectory: '${{ parameters.buildDirectory }}'
+                          script: |
+                            Coverage_Report_Artifact_Name="Code Coverage Report"
```

Best practice for bash scripts, to make sure any error while executing any line immediately terminates the whole script and thus the ADO step with a failure code.
```suggestion
                            set -eux -o pipefail
                            Coverage_Report_Artifact_Name="Code Coverage Report"
```